### PR TITLE
feat: rename 'assert-arg-order' rule to 'no-invalid-debug-function-arguments' to also detect invalid usages of 'deprecate' and 'warn'.

### PR DESCRIPTION
**Pull request by bmish**
Renames the `assert-arg-order` rule to `no-invalid-debug-function-arguments`. Generalizes it to detect the `assert` argument order mistake along with two other similar functions (`deprecate` and `warn`).

[Requested](https://github.com/ember-cli/eslint-plugin-ember/pull/362#issuecomment-450131918) by @Turbo87.

### CODE REVIEW

**Comment by bmish** [comment]
@rwjblue good idea, I renamed the rule to `no-invalid-debug-arguments`.

File: docs/rules/no-invalid-debug-function-arguments.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```js
+import { assert, warn } from '@ember/debug';
+import { deprecate } from '@ember/application/deprecations';
+
+...
+
+assert(label, 'Label must be present.');
+warn(label, 'Label must be present.');
+deprecate(title, 'Title is no longer supported.');
```
Comment (Turbo87) [inline docs/rules/no-invalid-debug-function-arguments.md]: @bmish I think `deprecate()` is supposed to always have a third argument (an options hash).

Comment (bmish) [reply docs/rules/no-invalid-debug-function-arguments.md]: Good point, added the third argument to both `warn` and `deprecate`.

**Comment by Turbo87** [comment]
thanks again @bmish!

one small potential improvement would be to not show the generic error message, but instead, show an error message mentioning the debug function that you actually used.

**Comment by bmish** [comment]
@Turbo87 done: https://github.com/ember-cli/eslint-plugin-ember/pull/365

**Pull request merged**